backend/src/test/scala/com/softwaremill/adopttapir/starter/StarterServiceITSpec.scala

```diff
@@ -119,7 +123,138 @@ class StarterServiceITSpec extends BaseTest {
   it should "return zip file containing working sbt folder with ZIO ZIOHttp implementation" in {
     val service = createStarterService
 
-    val starterDetails = StarterDetails("projectName", "com.softwaremill", ZIOEffect, ZIOHttp, defaultTapirVersion)
+    val starterDetails =
+      StarterDetails("projectName", "com.softwaremill", ZIOEffect, ZIOHttp, defaultTapirVersion, documentationAdded = false)
+
+    IO.blocking(BFile.newTemporaryDirectory("sbtTesting"))
+      .bracket { tempDir =>
+        for {
+          zipFile <- service.generateZipFile(starterDetails).map(_.toScala)
+          _ <- IO.blocking {
+            zipFile.unzipTo(tempDir)
+            zipFile.delete()
+          }
+          _ <- IO.blocking(os.proc("sbt", ";compile ;test").call(cwd = os.Path(tempDir.toJava)).exitCode shouldBe 0)
+
+        } yield zipFile
+
+      }(tempDir => IO.blocking(tempDir.delete()))
+      .unsafeRunSync()
+  }
+
+  it should "return zip file containing working sbt folder with Future Akka implementation with doc endpoint" in {
+    val service = createStarterService
+
+    val starterDetails =
+      StarterDetails("projectName", "com.softwaremill", FutureEffect, Akka, defaultTapirVersion, documentationAdded = true)
+
+    IO.blocking(BFile.newTemporaryDirectory("sbtTesting"))
```

Comment:
[minor] can be probably extracted to a helper method :)

Reply:
if the tests are the same but vary in parameters then we should extrat the common parts, much easier to read

Reply:
but looks much bette r:)

Reply:
well if you call it `sbtCompileTest(tempDir)` it will be obvious as well and the number of strings will decrease ;)

Reply:
In that particular case when all tests are different only in request I think Your idea will be better. I will hide test method back to helper function.